core/src/subgraph/instance_manager.rs

```diff
@@ -331,6 +331,13 @@ impl<S: SubgraphStore> SubgraphInstanceManager<S> {
             })
             .collect();
 
+        let max_end_block: Option<BlockNumber> = if manifest.data_sources.len() == end_blocks.len()
```

Comment:
A subgraph can also have offchain data sources, so we cant just compare it like this since offchain datasources dont have an end block